### ISSUE TITLE
sodium_compare() is broken

### ISSUE THREAD

**Issue by thebigb** [jedisct1/libsodium#313]
[Test case gist](https://gist.github.com/thebigb/d234af4651afb59cfc40)
```
   value  high   low
a:     0     0     0
b:   511     1   255
c: 65281   255     1

a < b: true
a < c: true
b < a: false
b < c: false <--- 
c < a: false
c < b: true <---
```
I'm pretty sure that the marked results are incorrect.

**Comment by jedisct1** [comment]
`sodium_compare()` works like `memcmp()` which is fine to compare numbers in big-endian notation, but `sodium_increment()` assumes little-endian, so they indeed don't play well with each other.

Looks like no public code is using `sodium_compare()` yet so I guess we should change it to use little-endian in the next version.

**Comment by thebigb** [comment]
@jedisct1

The [documentation](http://doc.libsodium.org/helpers/index.html#comparing-large-numbers) and commit message (https://github.com/jedisct1/libsodium/commit/d667efde6863528d77711ed6db0d89de92b59d58) imply that it already should be little-endian comparison.

**Comment by jedisct1** [comment]
Yes, that was the intent. This is being fixed.

**Comment by jedisct1** [comment]
Fixed in 84695c8d426cdf9ce7286b3e191625408b66aab5, thanks a ton for reporting this!
I'll tag a new version shortly.

**Comment by thebigb** [comment]
Cheers, thanks for the quick response/fix!
